src/geometry/face.rs

```diff
@@ -88,24 +90,32 @@ impl OrientedBlockFace {
         [self.signed_normal().as_vec3().to_array(); 4]
     }
 
-    /// Returns the 6 vertex indices for the quad in order to make two triangles in a mesh. Winding order depends on both the
-    /// sign of the surface normal and the permutation of the UVs.
+    /// Returns the 6 vertex indices for the quad in order to make two triangles
+    /// in a mesh. Winding order depends on both the sign of the surface normal
+    /// and the permutation of the UVs.
+    ///
+    /// Front faces will be wound counterclockwise, and back faces clockwise, as
+    /// per convention.
     #[inline]
     pub fn quad_mesh_indices(&self, start: u32) -> [u32; 6] {
         quad_indices(start, self.n_sign * self.permutation.sign() > 0)
     }
 
-    /// Returns the UV coordinates of the 4 corners of the quad. Returns vertices in the same order as
-    /// `OrientedBlockFace::quad_corners`.
+    /// Returns the UV coordinates of the 4 corners of the quad. Returns
+    /// vertices in the same order as [`OrientedBlockFace::quad_corners`].
     ///
-    /// `u_flip_face` should correspond to the field on [`QuadCoordinateConfig`](crate::QuadCoordinateConfig). See the docs
+    /// `u_flip_face` should correspond to the field on
+    /// [`QuadCoordinateConfig`](crate::QuadCoordinateConfig). See the docs
     /// there for more info.
     ///
-    /// This is just one way of assigning UVs to voxel quads. It assumes that each material has a single tile texture with
-    /// wrapping coordinates, and each voxel face should show the entire texture. It also assumes a particular orientation for
-    /// the texture. This should be sufficient for minecraft-style meshing.
+    /// This is just one way of assigning UVs to voxel quads. It assumes that
+    /// each material has a single tile texture with wrapping coordinates, and
+    /// each voxel face should show the entire texture. It also assumes a
+    /// particular orientation for the texture. This should be sufficient for
+    /// minecraft-style meshing.
```

Comment:
> This should be sufficient for minecraft-style meshing.

I wonder has this held true for brine, seeing as it is one of the most complete Minecraft clients I've seen. Not sure if you're still using block-mesh, but I appreciate this contribution.

Reply:
I actually didn't understand this statement fully. What is the assumption made about texture orientation? is it regarding how the top/bottom faces' textures are oriented?

I currently am not using block-mesh for the chunk meshing, so I'm not positive if the assumption held true. But I think I will use it to generate item meshes (like the diamond pickaxe you hold in your hand, that model needs to be generated from the pixels of the item texture).